controllers/overlay.go

```diff
@@ -0,0 +1,82 @@
+package controllers
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/streampets/backend/models"
+	"github.com/streampets/backend/services"
+)
+
+type viewerParams struct {
+	ChannelID models.UserID
+	OverlayID uuid.UUID
+}
+
+type channelService interface {
```

Comment:
I feel like there must be some **Go** advanced reasoning for doing this, which I'm super curious about since this is the first time I've actually looked at Go, it still looks odd having these two interfaces here solely for typing the `OverlayController`.

Reply:
Definitely an anti pattern in other languages but in Go you "accept interfaces and return structs"